### PR TITLE
Add view duplication

### PR DESCRIPTION
**Pull request by grimm**

### CODE REVIEW

**Comment by grimm** [comment]
Ready for review, except the rubocop error which might need to be fixed in the rubocop settings?

File: app/controllers/views_controller.rb [hunk 1]
```diff
@@ -65,6 +65,23 @@ def destroy
     end
   end
 
+  def duplicate
+    new_view = @workspace.views.new(name: @view.name + '-duplicate', center_lat: @view.center_lat, center_lng: @view.center_lng, zoom: @view.zoom, basemap: @view.basemap)
```
Comment (teknofire) [inline app/controllers/views_controller.rb]: abstract this functionality into a method on the View model

**Comment by grimm** [comment]
Ready for review.

File: app/models/view.rb [hunk 2]
```diff
@@ -12,4 +12,15 @@ class View < ApplicationRecord
   def to_s
     name
   end
+
+  def duplicate
+    new_view = workspace.views.new(name: name + '-duplicate', center_lat: center_lat, center_lng: center_lng, zoom: zoom, basemap: basemap)
```
Comment (teknofire) [inline app/models/view.rb]: When using the relation should use `build` instead of `new`.  `new` does not create the relation to the parent object.

**Pull request merged**